test/integration/controller/mpijob/mpijob_controller_test.go

```diff
@@ -561,4 +561,154 @@ var _ = ginkgo.Describe("Job controller interacting with scheduler", func() {
+				Request(kubeflow.MPIReplicaTypeLauncher, corev1.ResourceCPU, "3").
+				Request(kubeflow.MPIReplicaTypeWorker, corev1.ResourceCPU, "4").
+				Obj()
+			lookupKey := types.NamespacedName{Name: job.Name, Namespace: job.Namespace}
+			createdJob := &kubeflow.MPIJob{}
+
+			nodeSelectors := func(j *kubeflow.MPIJob) map[kubeflow.MPIReplicaType]map[string]string {
+				ret := map[kubeflow.MPIReplicaType]map[string]string{}
+				for k := range j.Spec.MPIReplicaSpecs {
+					ret[k] = j.Spec.MPIReplicaSpecs[k].Template.Spec.NodeSelector
+				}
+				return ret
+			}
+
+			ginkgo.By("create a job", func() {
+				gomega.Expect(k8sClient.Create(ctx, job)).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("job should be suspend", func() {
+				gomega.Eventually(func() *bool {
+					gomega.Expect(k8sClient.Get(ctx, lookupKey, createdJob)).Should(gomega.Succeed())
+					return createdJob.Spec.RunPolicy.Suspend
+				}, util.Timeout, util.Interval).Should(gomega.Equal(pointer.Bool(true)))
+			})
+
+			// backup the the node selectors
+			originalNodeSelectors := nodeSelectors(createdJob)
+
+			ginkgo.By("create a localQueue", func() {
+				gomega.Expect(k8sClient.Create(ctx, localQueue)).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("job should be unsuspended", func() {
+				gomega.Eventually(func() *bool {
+					gomega.Expect(k8sClient.Get(ctx, lookupKey, createdJob)).Should(gomega.Succeed())
+					return createdJob.Spec.RunPolicy.Suspend
+				}, util.Timeout, util.Interval).Should(gomega.Equal(pointer.Bool(false)))
+			})
+
+			ginkgo.By("the node selectors should be updated", func() {
+				gomega.Eventually(func() map[kubeflow.MPIReplicaType]map[string]string {
+					gomega.Expect(k8sClient.Get(ctx, lookupKey, createdJob)).Should(gomega.Succeed())
+					return nodeSelectors(createdJob)
+				}, util.Timeout, util.Interval).ShouldNot(gomega.Equal(originalNodeSelectors))
+			})
+
+			ginkgo.By("delete the localQueue to prevent readmission", func() {
+				gomega.Expect(util.DeleteLocalQueue(ctx, k8sClient, localQueue)).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("delete the workload to stop the job", func() {
+				wl := &kueue.Workload{}
+				wlKey := types.NamespacedName{Name: workloadmpijob.GetWorkloadNameForMPIJob(job.Name), Namespace: job.Namespace}
+				gomega.Expect(k8sClient.Get(ctx, wlKey, wl)).Should(gomega.Succeed())
+				gomega.Expect(util.DeleteWorkload(ctx, k8sClient, wl)).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("the node selectors should be restored", func() {
+				gomega.Eventually(func() map[kubeflow.MPIReplicaType]map[string]string {
+					gomega.Expect(k8sClient.Get(ctx, lookupKey, createdJob)).Should(gomega.Succeed())
+					return nodeSelectors(createdJob)
+				}, util.Timeout, util.Interval).Should(gomega.Equal(originalNodeSelectors))
+			})
+		})
+
+		ginkgo.It("Should continue to use the admission node selectors if the annotation is corrupted", func() {
```

Comment:
nit: I would suggest deleting this test case as it shows a scenario which is blocked by webhooks.

Reply:
done